cmd/deviceWSConnector.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020-2021 JackTrip Labs, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/jacktrip/jacktrip-agent/pkg/client"
+)
+
+type webSocketConnector struct {
+	Conn          *websocket.Conn
+	Mu            sync.Mutex
+	APIOrigin     string
+	IsInitialized bool
+}
+
+// Connector is used as a singleton object of webSocketConnector
+var Connector webSocketConnector
+
+// ConfigChannel holds config objects that came through a websocket connection and is consumed by a handler in device.go
+var ConfigChannel = make(chan client.AgentConfig, 100)
+
+// ReceivePingChannel is used to control ReceivePingHandler Goroutine
+var ReceivePingChannel chan bool
```

Comment:
similar to devicePinger, i think we can (and should) eliminate the singletons by moving the variables & methods into the `webSocketConnector` struct.